### PR TITLE
CVSB-3887

### PR DESCRIPTION
**Pull request by mgherceanu**

### CODE REVIEW

File: GoogleService-Info.plist [hunk 1]
```diff
@@ -1 +0,0 @@
-
```
Comment (vladddragomir) [inline GoogleService-Info.plist]: revert this please

File: src/app/app.component.ts [hunk 2]
```diff
@@ -32,12 +32,12 @@ export class MyApp {
...
               private mobileAccessibility: MobileAccessibility,
               private renderer: Renderer2,
-              private firebase: Firebase,
+              private firebaseLogsService: FirebaseLogsService,
               private screenOrientation: ScreenOrientation) {
     platform.ready().then(() => {
       statusBar.overlaysWebView(true);
       statusBar.styleLightContent();
-      this.firebase.logEvent('test_event', {content_type: 'page_view', item_id: 'open_app'});
+      this.firebaseLogsService.logEvent('test_event', 'page_view', 'open_app');
```
Comment (vladddragomir) [inline src/app/app.component.ts]: please add this to app enums

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 3]
```diff
@@ -46,6 +51,12 @@ export class OdometerReadingPage {
   }
 
   onSave() {
+    this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_end_time = Date.now();
+    this.firebaseLogsService.logEvent('add_odometer_reading_time_taken', 'add_odometer_reading_end_time', this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_end_time.toString());
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: please add this to app enums

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 4]
```diff
@@ -46,6 +51,12 @@ export class OdometerReadingPage {
...
+
+    this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_time_taken = this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_end_time - this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_start_time;
+    this.firebaseLogsService.logEvent('add_odometer_reading_time_taken', 'add_odometer_reading_time_taken', this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_time_taken.toString());
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: same here

File: src/pages/testing/test-creation/test-cancel/test-cancel.ts [hunk 5]
```diff
@@ -102,7 +102,8 @@ export class TestCancelPage {
         () => {
           LOADING.dismiss();
           TRY_AGAIN_ALERT.present();
-          this.firebase.logEvent('test_error', {content_type: 'error', item_id: "Test submission failed"});
+          this.firebaseLogsService.logEvent('test_error', 'error', 'Test submission failed');
+          // this.firebase.logEvent('test_error', {content_type: 'error', item_id: "Test submission failed"});
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-cancel/test-cancel.ts]: same here and also remove the comment

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 6]
```diff
@@ -162,6 +162,10 @@ export class TestCreatePage implements OnInit {
   }
 
   onOdometer(index: number) {
+    this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_start_time = Date.now();
+    this.firebaseLogsService.logEvent('add_odometer_reading_time_taken', 'add_odometer_reading_start_time', this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_start_time.toString());
+    // this.firebase.logEvent('add_odometer_reading_time_taken', {content_type: 'add_odometer_reading_start_time', item_id: this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_start_time.toString()});
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-create/test-create.ts]: same

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 7]
```diff
@@ -244,7 +248,8 @@ export class TestCreatePage implements OnInit {
         buttons: [APP_STRINGS.OK]
       });
       alert.present();
-      this.firebase.logEvent('test_error', {content_type: 'error', item_id: "Not all tests completed before review"});
+      this.firebaseLogsService.logEvent('test_error', 'error', 'Not all tests completed before review');
+      // this.firebase.logEvent('test_error', {content_type: 'error', item_id: "Not all tests completed before review"});
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-create/test-create.ts]: same

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 8]
```diff
@@ -253,7 +258,8 @@ export class TestCreatePage implements OnInit {
         buttons: [APP_STRINGS.OK]
       });
       alert.present();
-      this.firebase.logEvent('test_error', {content_type: 'error', item_id: "No test added before review"});
+      this.firebaseLogsService.logEvent('test_error', 'error', 'No test added before review');
+      // this.firebase.logEvent('test_error', {content_type: 'error', item_id: "No test added before review"});
```
Comment (vladddragomir) [inline src/pages/testing/test-creation/test-create/test-create.ts]: same

File: src/pages/testing/vehicle/add-preparer/add-preparer.ts [hunk 9]
```diff
@@ -149,4 +151,17 @@ export class AddPreparerPage implements OnInit {
     }
     return this.authService.hasRights(userRights, neededRights);
   }
+
+  logIntoFirebase() {
+    this.firebaseLogsService.confirm_preparer_time.confirm_preparer_end_time = Date.now();
+    this.firebaseLogsService.logEvent('confirm_preparer_time_taken', 'confirm_preparer_end_time', this.firebaseLogsService.confirm_preparer_time.confirm_preparer_end_time.toString());
+    // this.firebase.logEvent('confirm_preparer_time_taken', {content_type: 'confirm_preparer_end_time', item_id: this.firebaseLogsService.confirm_preparer_time.confirm_preparer_end_time.toString()});
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/add-preparer/add-preparer.ts]: same

File: src/pages/testing/vehicle/add-preparer/add-preparer.ts [hunk 10]
```diff
@@ -149,4 +151,17 @@ export class AddPreparerPage implements OnInit {
...
+
+    this.firebaseLogsService.confirm_preparer_time.confirm_preparer_time_taken = this.firebaseLogsService.confirm_preparer_time.confirm_preparer_end_time - this.firebaseLogsService.confirm_preparer_time.confirm_preparer_start_time;
+    this.firebaseLogsService.logEvent('confirm_preparer_time_taken', 'confirm_preparer_time_taken', this.firebaseLogsService.confirm_preparer_time.confirm_preparer_time_taken.toString());
+    // this.firebase.logEvent('confirm_preparer_time_taken', {
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/add-preparer/add-preparer.ts]: same

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts [hunk 11]
```diff
@@ -36,6 +38,20 @@ export class VehicleDetailsPage {
     this.viewCtrl.setBackButtonText(this.fromTestCreatePage ? APP_STRINGS.TEST : APP_STRINGS.IDENTIFY_VEHICLE);
   }
 
+  ionViewDidEnter() {
+    this.firebaseLogsService.search_vehicle_time.search_vehicle_end_time = Date.now();
+    this.firebaseLogsService.logEvent('search_vehicle_time_taken', 'search_vehicle_end_time', this.firebaseLogsService.search_vehicle_time.search_vehicle_end_time.toString());
+    // this.firebase.logEvent('search_vehicle_time_taken', {content_type: 'search_vehicle_end_time', item_id: this.firebaseLogsService.search_vehicle_time.search_vehicle_end_time.toString()});
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: same

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts [hunk 12]
```diff
@@ -36,6 +38,20 @@ export class VehicleDetailsPage {
...
+
+    this.firebaseLogsService.search_vehicle_time.search_vehicle_time_taken = this.firebaseLogsService.search_vehicle_time.search_vehicle_end_time - this.firebaseLogsService.search_vehicle_time.search_vehicle_start_time;
+    this.firebaseLogsService.logEvent('search_vehicle_time_taken', 'search_vehicle_time_taken', this.firebaseLogsService.search_vehicle_time.search_vehicle_time_taken.toString());
+    // this.firebase.logEvent('search_vehicle_time_taken', {content_type: 'search_vehicle_time_taken', item_id: this.firebaseLogsService.search_vehicle_time.search_vehicle_time_taken.toString()});
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: same

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts [hunk 13]
```diff
@@ -36,6 +38,20 @@ export class VehicleDetailsPage {
...
+
+    this.firebaseLogsService.confirm_vehicle_time.confirm_vehicle_start_time = Date.now();
+    this.firebaseLogsService.logEvent('confirm_vehicle_time_taken', 'confirm_vehicle_start_time', this.firebaseLogsService.confirm_vehicle_time.confirm_vehicle_start_time.toString());
+    // this.firebase.logEvent('confirm_vehicle_time_taken', {content_type: 'confirm_vehicle_start_time', item_id: this.firebaseLogsService.confirm_vehicle_time.confirm_vehicle_start_time.toString()});
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: same

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts [hunk 14]
```diff
@@ -100,4 +118,15 @@ export class VehicleDetailsPage {
       }
     )
   }
+
+  loggingInAlertHandler() {
+    this.firebaseLogsService.confirm_vehicle_time.confirm_vehicle_end_time = Date.now();
```
Comment (vladddragomir) [inline src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: add strings to app enums

File: src/pages/visit/visit-timeline/visit-timeline.ts [hunk 15]
```diff
@@ -62,8 +62,11 @@ export class VisitTimelinePage implements OnInit {
   }
 
   createNewTestReport(): void {
+    this.firebaseLogsService.search_vehicle_time.search_vehicle_start_time = Date.now();
+    this.firebaseLogsService.logEvent('search_vehicle_time_taken', 'search_vehicle_start_time', this.firebaseLogsService.search_vehicle_time.search_vehicle_start_time.toString());
+    // this.firebase.logEvent('search_vehicle_time_taken', {content_type: 'search_vehicle_start_time', item_id: this.firebaseLogsService.search_vehicle_time.search_vehicle_start_time.toString()});
```
Comment (vladddragomir) [inline src/pages/visit/visit-timeline/visit-timeline.ts]: same

File: src/pages/visit/visit-timeline/visit-timeline.ts [hunk 16]
```diff
@@ -85,10 +88,11 @@ export class VisitTimelinePage implements OnInit {
         this.storageService.delete(STORAGE.STATE);
         this.visitService.visit = {} as VisitModel;
         LOADING.dismiss();
-        this.navCtrl.push('EndVisitConfirmPage', {testStationName: this.visit.testStationName});
+        this.navCtrl.push(PAGE_NAMES.END_VISIT_CONFIRM_PAGE, {testStationName: this.visit.testStationName});
       }, (error) => {
         LOADING.dismiss();
-        this.firebase.logEvent('test_error', {content_type: 'error', item_id: "Ending activity failed"});
+        this.firebaseLogsService.logEvent('test_error', 'error', 'Ending activity failed');
+        // this.firebase.logEvent('test_error', {content_type: 'error', item_id: "Ending activity failed"});
```
Comment (vladddragomir) [inline src/pages/visit/visit-timeline/visit-timeline.ts]: same

File: GoogleService-Info.plist (hunk 1 above)
Comment (mgherceanu) [reply GoogleService-Info.plist]: done

File: src/app/app.component.ts (hunk 2 above)
Comment (mgherceanu) [reply src/app/app.component.ts]: done

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts (hunk 3 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: done

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts (hunk 4 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: done

File: src/pages/testing/test-creation/test-cancel/test-cancel.ts (hunk 5 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-cancel/test-cancel.ts]: DONE

File: src/pages/testing/test-creation/test-create/test-create.ts (hunk 6 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-create/test-create.ts]: DONE

File: src/pages/testing/test-creation/test-create/test-create.ts (hunk 7 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-create/test-create.ts]: DONE

File: src/pages/testing/test-creation/test-create/test-create.ts (hunk 8 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-create/test-create.ts]: DONE

File: src/pages/testing/vehicle/add-preparer/add-preparer.ts (hunk 9 above)
Comment (mgherceanu) [reply src/pages/testing/vehicle/add-preparer/add-preparer.ts]: DONE

File: src/pages/testing/vehicle/add-preparer/add-preparer.ts (hunk 10 above)
Comment (mgherceanu) [reply src/pages/testing/vehicle/add-preparer/add-preparer.ts]: DONE

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts (hunk 11 above)
Comment (mgherceanu) [reply src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: DONE

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts (hunk 13 above)
Comment (mgherceanu) [reply src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: DONE

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts (hunk 14 above)
Comment (mgherceanu) [reply src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: DONE

File: src/pages/testing/vehicle/vehicle-details/vehicle-details.ts (hunk 12 above)
Comment (mgherceanu) [reply src/pages/testing/vehicle/vehicle-details/vehicle-details.ts]: DONE

File: src/pages/visit/visit-timeline/visit-timeline.ts (hunk 15 above)
Comment (mgherceanu) [reply src/pages/visit/visit-timeline/visit-timeline.ts]: DONE

File: src/pages/visit/visit-timeline/visit-timeline.ts (hunk 16 above)
Comment (mgherceanu) [reply src/pages/visit/visit-timeline/visit-timeline.ts]: done

**Comment by agoncearenco** [comment]
The following lines will create 2 separate events with 1 parameter each. Isn't it better if we will have one single event in this case?
https://github.com/dvsa/cvs-app-mobile/blob/47104adcf5941fea950a8b1aac70178a33eeee21/src/pages/testing/vehicle/vehicle-details/vehicle-details.ts#L42-L46

File: src/pages/testing/test-creation/odometer-reading/odometer-reading.ts [hunk 17]
```diff
@@ -46,6 +51,12 @@ export class OdometerReadingPage {
   }
 
   onSave() {
+    this.firebaseLogsService.add_odometer_reading_time.add_odometer_reading_end_time = Date.now();
```
Comment (agoncearenco) [inline src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: Shouldn't this parameter be logged to Firebase with the _HH:MM:SS_ format as described [here](https://jira.i-env.net/browse/CVSB-3887#descriptionmodule_heading)?

Comment (mgherceanu) [reply src/pages/testing/test-creation/odometer-reading/odometer-reading.ts]: i have changed this to be in HH:MM:SS format, but only the third parameter is required in this format.

**Pull request closed without merging**